lib/jekyll/filters.rb

```diff
@@ -389,7 +389,11 @@ def groupable?(element)
     private
     def item_property(item, property)
       if item.respond_to?(:to_liquid)
-        item.to_liquid[property.to_s]
+        res = item.to_liquid
+        property.to_s.split(".").each do |attr|
+          res = res[attr]
+        end
+        res
       elsif item.respond_to?(:data)
         item.data[property.to_s]
```

Comment:
Yeah I would like to but in the `test/test_filters.rb` file I can't figure out how can I give the input as an object responding to `data`.

I looked for other filters and there is no filters testing for data inputs.